### ISSUE TITLE
Encoding bug with accent characters

### ISSUE THREAD

**Issue by copydepo** [meilisearch/MeiliSearch#1631]
Searching with hungarian characters (á, é, ó, ö, ő, ü, ű) doesn't work as expected. The highlight does not highlight these letters in search results. I assume it counts it as a typo also.

**Steps to reproduce**
1. Searching for keyword "nyomtató"
2. Results highlighting only "nyomtat" without the ending ó.

**Comment by curquiza** [comment]
Hello @copydepo

Which version of MeiliSearch do you use?

I've tested to add this doc:

```json
[
    {
        "id": 1, "title": "nyomtató"
    }
]
```

and then search for 

```json
{
  "q": "nyomtató",
  "attributesToHighlight": ["*"]
}
```

and I get correctly:

```json
{
    "hits": [
        {
            "id": 1,
            "title": "nyomtató",
            "_formatted": {
                "id": 1,
                "title": "<em>nyomtató</em>"
            }
        }
    ],
    "nbHits": 1,
    "exhaustiveNbHits": false,
    "query": "nyomtató",
    "limit": 20,
    "offset": 0,
    "processingTimeMs": 5
}
```

**Comment by copydepo** [comment]
Hello @curquiza,

thanks for fast response.

The version is 0.20.0 and here is a picture:
![image](https://user-images.githubusercontent.com/78180245/130806657-f4a0d852-0fe2-4c01-b6b5-320f7fa3e379.png)

**Comment by CaroFG** [comment]
Hi! I tried to reproduce the issue and I found out that the problem is that when using accents, MeiliSearch highlights one less character.

**Steps to reproduce the issue** 
1. Add the movies dataset to MeiliSearch
2. Update the movie entitled "Lock, Stock and Two Smoking Barrels"  and change the title to "Lőck, Stóck and Two Smöking Barrels"
3. Wait for the update to be processed and search for "Lőck"
4.  The result only highlights "Lőc"

![Capture d’écran 2021-08-25 à 16 16 23](https://user-images.githubusercontent.com/48251481/130807316-880bd309-d34f-4767-b258-b6f3e2732702.png)
